packages/axelar-wasm-std/src/chain_name.rs

```diff
@@ -0,0 +1,461 @@
+use std::fmt;
+use std::fmt::Display;
+use std::str::FromStr;
+
+use const_str::contains;
+use cosmwasm_schema::cw_serde;
+use cosmwasm_std::{StdError, StdResult};
+use cw_storage_plus::{Key, KeyDeserialize, Prefixer, PrimaryKey};
+use error_stack::report;
+use valuable::Valuable;
+
+use crate::{FnExt, FIELD_DELIMITER};
+
+#[derive(thiserror::Error, Debug, PartialEq)]
+pub enum Error {
+    #[error("chain name is invalid")]
+    InvalidChainName,
+}
+
+impl From<Error> for crate::error::ContractError {
+    fn from(error: Error) -> Self {
+        report!(error).into()
+    }
+}
```

Comment:
```suggestion
use crate as axelar_wasm_std;
use crate::{FnExt, FIELD_DELIMITER};

#[derive(thiserror::Error, Debug, PartialEq, IntoContractError)]
pub enum Error {
    #[error("chain name is invalid")]
    InvalidChainName,
}
```

Reply:
can't use stuff from `axelar-wasm-std-derive` because it's only available to `axelar-wasm-std` using the `derive` feature flag, it would need to depend on itself using the `derive` feature flag